test/test_functional.py

```diff
@@ -1,90 +1,78 @@
 import torch
 from torchtext import functional
+from .common.parameterized_utils import nested_params
 
 from .common.torchtext_test_case import TorchtextTestCase
 
-
 class TestFunctional(TorchtextTestCase):
-    def _to_tensor(self, test_scripting):
-        input = [[1, 2], [1, 2, 3]]
-        padding_value = 0
 
+    @nested_params(
+        [True, False],
+        [
+            [[[1, 2], [1, 2, 3]], 0, [[1, 2, 0], [1, 2, 3]]],
+            [[[1, 2], [1, 2, 3]], 1, [[1, 2, 1], [1, 2, 3]]],
+            [[1, 2], 0, [1, 2]],
+        ]
+    )
+    def test_to_tensor(self, test_scripting, configs):
+        """test tensorization on both single sequence and batch of sequence"""
+        inputs, padding_value, expected_list = configs
         func = functional.to_tensor
         if test_scripting:
             func = torch.jit.script(func)
-        actual = func(input, padding_value=padding_value)
-        expected = torch.tensor([[1, 2, 0], [1, 2, 3]], dtype=torch.long)
-        torch.testing.assert_close(actual, expected)
+        else:
+            with self.assertRaises(TypeError):
+                func("test")
```

Comment:
Just wondering if this additional assertion helps to increase test coverage by ensuring the exception handling implementation is working correctly?

Reply:
Yes, this is to test that an input type that is not a `List[int]` or `List[List[int]]` raises a `TypeError` (see [here](https://github.com/pytorch/text/blob/main/torchtext/functional.py#L37)).

Reply:
Looking at the flow of the test logic, this assertion kind of showed us suddenly. 

Also that assertion does not require the parameterized input, so why not put it in a stand-alone test?

Reply:
@mthrok thanks for the comments! I addressed all the issues in #1667.